packages/app-server-issuer/src/exchange-provider/ExchangeProviderManager.ts

```diff
@@ -0,0 +1,136 @@
+import { number, bytes } from '@ckb-lumos/codec';
+import { Cell, Indexer, helpers, HexString, BI as BigNumber } from '@ckb-lumos/lumos';
+import { AcpTransferSudtBuilder } from '@ckitjs/ckit';
+import { ServerContext } from '../types';
+import { Config } from './config';
+
+export class ExchangeProviderManager {
+  private cells: Cell[] = [];
+  private initiated: boolean;
+  private config!: Config;
+  private context!: ServerContext;
+
+  private static sigleton: ExchangeProviderManager | undefined;
+
+  static getInstance(): ExchangeProviderManager {
+    if (ExchangeProviderManager.sigleton == undefined) {
+      ExchangeProviderManager.sigleton = new ExchangeProviderManager();
+    }
+
+    return ExchangeProviderManager.sigleton;
+  }
+
+  private constructor() {
+    this.initiated = false;
+  }
+
+  assertInitiated(): void {
+    if (!this.initiated) {
+      throw new Error('ExchangeProviderManager is not initiated');
+    }
+  }
+
+  private async refreshCells(): Promise<void> {
+    this.assertInitiated();
+
+    const indexer = new Indexer(this.context.ckitProvider.mercuryUrl, this.context.ckitProvider.rpcUrl);
+    const cells = (
+      await indexer.getCells({
+        script_type: 'lock',
+        script: this.context.ckitProvider.parseToScript(this.context.exchangeSigner!.getAddress()),
+        filter: {
+          script: this.context.ckitProvider.newScript('SUDT', this.config.sudtArgs),
+        },
+      })
+    ).objects;
+
+    if (cells.length < this.config.exchangeCellCount) {
+      await this.createProviderCells(this.config.exchangeCellCount - cells.length);
+      await this.refreshCells();
+    } else {
+      this.cells = cells.sort((a, b) => BigNumber.from(b.block_number).sub(BigNumber.from(a.block_number)).toNumber());
+    }
+  }
+
+  private async createProviderCells(amount: number): Promise<void> {
+    this.assertInitiated();
+
+    const builder = new AcpTransferSudtBuilder(
+      {
+        allowDuplicateRecipient: true,
+        recipients: Array.from({ length: amount }).map(() => {
+          return {
+            recipient: this.context.exchangeSigner!.getAddress(),
+            sudt: this.context.ckitProvider.newScript('SUDT', this.config.sudtArgs),
+            amount: bytes.hexify(number.Uint128.pack(0)),
+            policy: 'createCell',
+            additionalCapacity: BigNumber.from(this.config.initCapacity).toHexString(),
+          };
+        }),
+      },
+      this.context.ckitProvider,
+      this.context.txSigner!.getAddress(),
+    );
+    try {
+      const unsignedTx = await builder.build();
+      const tx = await this.context.txSigner.seal(unsignedTx);
+      const hash = await this.context.ckitProvider.sendTransaction(tx);
+      await this.context.ckitProvider.waitForTransactionCommitted(hash);
+    } catch (e) {
+      console.log(e);
```

Comment:
```suggestion
      throw new Error(`init provider cell failed, please try again later`);
```

Reply:
Here I want to print error to the screen, so I will use `logger` instead of using `console.log` directly.